File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int ans=0;
        int n=nums.size();
        int l=0;
        int r=0;
        while(r<n){
            if(nums[r]==0){
                if(k==0){
                    while(nums[l]!=0) l+=1;
                    l+=1;

                }else{
                    k-=1;

                }
            }
            ans=max(ans,r-l+1);
            r++;
        }
        return ans;
    }
};

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int l = 0;
        int zeroes = 0;
        int maxlen = 0;
        int n = nums.length;

        for (int r=0;r < nums.length;r++) {
            if (nums[r] == 0) {
                zeroes++;
            }
            if (zeroes > k) {
                if (nums[l] == 0) {
                    zeroes--;
                }
                l++;
            }
            maxlen = Math.max(maxlen, r - l + 1);
        }
        return maxlen;
    }
}
